uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-requests.js

```diff
@@ -9,12 +10,453 @@ define([
+          vm.absenceTypes = absenceTypes;
+        });
+    }
+
+    /**
+     * Loads all requests
+     *
+     * @param {int} page - page number
+     * @return {Promise}
+     */
+    function loadAllRequests(page) {
+      vm.pagination.page = page;
+      vm.loading.content = true;
+      Contact.all(contactFilters(), {
+        page: 1,
+        size: 0
+      })
+        .then(function (users) {
+          vm.filteredUsers = users.list;
+
+          $q.all([
+            loadLeaveRequest('table'),
+            loadLeaveRequest('filter')
+          ])
+            .then(function () {
+              vm.loading.content = false;
+            })
+        });
+    }
+
+    /**
+     * Loads the departments option values
+     *
+     * @return {Promise}
+     */
+    function loadDepartments() {
+      return OptionGroup.valuesOf('hrjc_department')
+        .then(function (departments) {
+          vm.departments = departments;
+        });
+    }
+
+    /**
+     * Loads all leave requests
+     *
+     * @param {string} type - load leave requests for the either the filter or the table
+     * @return {Promise}
+     */
+    function loadLeaveRequest(type) {
+      var filterByStatus = type !== 'filter',
+        pagination = type === 'filter' ? {} : vm.pagination,
+        returnFields = type === 'filter' ? {
+            return: ['status_id']
+          } : {};
+
+      return LeaveRequest.all(leaveRequestFilters(filterByStatus), pagination, null, returnFields)
+        .then(function (leaveRequests) {
+          vm.leaveRequests[type] = leaveRequests;
+        });
+    }
+
+    /**
+     * Loads the level types option values
+     *
+     * @return {Promise}
+     */
+    function loadLevelTypes() {
+      return OptionGroup.valuesOf('hrjc_level_type')
+        .then(function (levels) {
+          vm.levelTypes = levels;
+        });
+    }
+
+    /**
+     * Loads the locations option values
+     *
+     * @return {Promise}
+     */
+    function loadLocations() {
+      return OptionGroup.valuesOf('hrjc_location')
+        .then(function (locations) {
+          vm.locations = locations;
+        });
+    }
+
+    /**
+     * Returns the filter object for leave request api
+     *
+     * @param {boolean} filterByStatus - if true then leave request api will be filtered using
+     * selected leave request status in the left navigation bar, which would be used to show the
+     * numbers of different statuses
+     * @return {Object}
+     */
+    function leaveRequestFilters(filterByStatus) {
+      var filters = vm.filters.leaveRequest;
+
+      return {
+        managed_by: vm.contactId,
+        type_id: filters.selectedAbsenceTypes ? filters.selectedAbsenceTypes.id : null,
+        status_id: prepareStatusFilter(filterByStatus),
+        from_date: {
+          from: filters.selectedPeriod.start_date
+        },
+        to_date: {
+          to: filters.selectedPeriod.end_date
+        },
+        contact_id: prepareContactID()
```

Comment:
As mentioned before, please group the one-liners together to improve readibility
```js
return {
  contact_id: prepareContactID(),
  managed_by: vm.contactId,
  status_id: prepareStatusFilter(filterByStatus),
  type_id: filters.selectedAbsenceTypes ? filters.selectedAbsenceTypes.id : null,
  from_date: {
    from: filters.selectedPeriod.start_date
  },
  to_date: {
    to: filters.selectedPeriod.end_date
  };
};
```